lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -303,8 +303,11 @@ sub _show {
     for my $key (OpenQA::Schema::Result::Jobs::SCENARIO_WITH_MACHINE_KEYS) {
         push(@conds, {"me.$key" => $job->get_column($key)});
     }
-    my $limit_previous = $self->param('limit_previous') // 10;    # arbitrary limit of previous results to show
-    my %attrs = (
+    # arbitrary limit of previous results to show
+    my $limit_previous = $self->param('limit_previous') // 10;
+    my %limit_query    = $job->scenario_hash;
+    my $limit_links    = join(' / ', map { ($limit_query{limit_previous} = $_) == $limit_previous ? "<b>$_</b>" : $self->link_to($_ => $self->url_with->query(%limit_query) . '#previous') } (10, 20, 50, 100, 400));
```

Comment:
can you pretty please write this in a readable way? I don't want to solve perl puzzles while scrolling forth and back